s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3Client.scala

```diff
@@ -664,6 +664,213 @@ final class S3Client(s3Settings: S3Settings, system: ActorSystem, mat: Materiali
                       sse: ServerSideEncryption): Sink[ByteString, CompletionStage[MultipartUploadResult]] =
     multipartUpload(bucket, key, ContentTypes.APPLICATION_OCTET_STREAM, CannedAcl.Private, MetaHeaders(Map()), sse)
 
+  /**
+   * Copy a S3 Object by making multiple requests.
+   *
+   * @param sourceBucket the source s3 bucket name
+   * @param sourceKey the source s3 key
+   * @param targetBucket the target s3 bucket name
+   * @param targetKey the target s3 key
+   * @param contentType an optional [[akka.http.javadsl.model.ContentType ContentType]]
+   * @param s3Headers any headers you want to add
+   * @param sse the server side encryption to use
+   * @return a [[java.util.concurrent.CompletionStage CompletionStage]] containing the [[akka.stream.alpakka.s3.javadsl.MultipartUploadResult MultipartUploadResult]] of the uploaded S3 Object
+   */
+  def multipartCopy(sourceBucket: String,
```

Comment:
Is it useful to have this many overloads? Users may have a hard time to pick the best overload for their purpose.
I see `putObject` does the same.

Reply:
I wasn't sure whether to add that many overloaded methods or, I saw `PutObject` is having overloaded methods so decided to add overloaded, I figured since  `Java ` doesn't have default parameters that might be reason overloaded methods were added in `PutObject`

Reply:
Maybe add something to the method names, so they can more easily be distinguished.
Or make sure good default values are available in the Java API and show those in the example code.